mods/ascent/machines/magnetotron.dm

```diff
@@ -0,0 +1,72 @@
+// Mutation device for turning alates into gynes.
+/obj/machinery/ascent_magnetotron
+	name = "\improper magnetotron"
+	desc = "An assemblage of alien machinery with large imposing dynamos."
+	color = COLOR_PURPLE
+	icon = 'icons/obj/stationobjs.dmi'
+	icon_state = "weightlifter"
+
+/obj/machinery/ascent_magnetotron/attack_hand(var/mob/user)
+	var/mob/living/carbon/human/target
+	for(var/mob/living/carbon/human/T in loc)
+		target = T
+		break
+
+	if(isnull(target))
+		to_chat(user, SPAN_WARNING("\icon[src] \The [src] chirps, \"No biological signature detected in [src].\""))
+		return TRUE
+
+	if(!isspecies(target, SPECIES_MANTID_ALATE))
+		to_chat(user, SPAN_WARNING("\icon[src] \The [src] chirps, \"Invalid biological signature detected. Safety mechanisms engaged -- only alates may go through metamorphosis.\""))
+		return TRUE
+	
+	if(!prob(100 / (get_total_gynes() + 1)))
+		// Oops it killed us.
+		target.visible_message(SPAN_DANGER("[target] shrieks loudly as \the [src] tears them apart."))
+		target.gib()
+		return TRUE
+
+	visible_message(SPAN_NOTICE("\icon[src] [src] begins to crackle and hum with energy as magnetic fields begin to fluctuate."))
```

Comment:
Might be worth putting this above the prob check since the machine is running one way or another.